RevenueCatUI/Templates/V2/Components/Stack/StackComponentView.swift

```diff
@@ -519,21 +519,32 @@ extension StackComponentViewModel {
         let validator = PackageValidator()
         let factory = ViewModelFactory()
         let offering = Offering(identifier: "", serverDescription: "", availablePackages: [])
+        let uiConfigProvider = UIConfigProvider(uiConfig: PreviewUIConfig.make())
 
         let viewModels = try component.components.map { component in
             try factory.toViewModel(
                 component: component,
                 packageValidator: validator,
                 offering: offering,
                 localizationProvider: localizationProvider,
-                uiConfigProvider: .init(uiConfig: PreviewUIConfig.make())
+                uiConfigProvider: uiConfigProvider
+            )
+        }
+
+        let badgeViewModels = try component.badge?.stack.value.components.map { component in
+            try factory.toViewModel(
+                component: component,
+                packageValidator: validator,
+                offering: offering,
+                localizationProvider: localizationProvider,
+                uiConfigProvider: uiConfigProvider
             )
         }
 
         try self.init(
             component: component,
             viewModels: viewModels,
-            badgeViewModels: [],
+            badgeViewModels: badgeViewModels ?? [],
```

Comment:
This is a convenience initialiser used only in previews. I ported the Badge previews to use real StackComponentView so I needed to actually parse the badge modifiers.